databases/core.py

```diff
@@ -363,7 +364,11 @@ async def commit(self) -> None:
         async with self._connection._transaction_lock:
             assert self._connection._transaction_stack[-1] is self
             self._connection._transaction_stack.pop()
-            await self._transaction.commit()
+            try:
+                await self._transaction.commit()
```

Comment:
(just a question) will the following work?
```python
try:
    transaction.commit()
except IntegrityError:
    transaction.rollback()
```
That is, catching it in the user code instead.

Reply:
> (just a question) will the following work?
> 
> ```python
> try:
>     transaction.commit()
> except IntegrityError:
>     transaction.rollback()
> ```
> 
> That is, catching it in the user code instead.

async def **rollback**(self) -> None:
        async with self._connection._transaction_lock:
            assert **self._connection._transaction_stack[-1]** is self  # Here _transaction_stack will be an empty list.
           ......

when I use **transaction.start()** before, unfortunately, TIDB did not support SAVEPOINT.
May I ask if you have any other suggestions? 

I look forward to your reply!